pallets/manta-pay/src/types.rs

```diff
@@ -184,7 +257,171 @@ impl TryFrom<SenderPost> for config::SenderPost {
+)]
+#[derive(Clone, Debug, Decode, Default, Encode, Eq, Hash, MaxEncodedLen, PartialEq, TypeInfo)]
+pub struct IncomingNote {
+    /// Ephemeral Public Key
+    pub ephemeral_public_key: Group,
+
+    /// Tag
+    pub tag: Tag,
+
+    /// Ciphertext
+    pub ciphertext: IncomingCiphertext,
+}
+
+impl From<v2::IncomingNote> for IncomingNote {
+    #[inline]
+    fn from(note: v2::IncomingNote) -> Self {
+        Self {
+            ephemeral_public_key: encode(note.ciphertext.ephemeral_public_key),
+            tag: encode(note.ciphertext.ciphertext.tag.0),
+            ciphertext: Array::from_iter(
+                note.ciphertext.ciphertext.message[0].0.iter().map(encode),
+            )
+            .into(),
+        }
+    }
+}
+
+impl TryFrom<IncomingNote> for v2::IncomingNote {
+    type Error = Error;
+
+    #[inline]
+    fn try_from(note: IncomingNote) -> Result<Self, Self::Error> {
+        Ok(Self {
+            header: EmptyHeader::default(),
+            ciphertext: hybrid::Ciphertext {
+                ephemeral_public_key: decode(note.ephemeral_public_key)?,
+                ciphertext: duplex::Ciphertext {
+                    tag: encryption::Tag(decode(note.tag)?),
+                    message: BlockArray(BoxArray(Box::new([CiphertextBlock(
+                        note.ciphertext
+                            .into_iter()
+                            .map(decode)
+                            .collect::<Result<Vec<_>, _>>()?
+                            .into(),
+                    )]))),
+                },
+            },
+        })
+    }
+}
+
+/// Full Incoming Note
+#[cfg_attr(
+    feature = "rpc",
+    derive(Deserialize, Serialize),
+    serde(crate = "manta_util::serde", deny_unknown_fields)
+)]
+#[derive(Clone, Debug, Decode, Default, Encode, Eq, Hash, MaxEncodedLen, PartialEq, TypeInfo)]
+pub struct FullIncomingNote {
+    /// Address Partition
+    pub address_partition: u8,
+
+    /// Incoming Note
+    pub incoming_note: IncomingNote,
+}
+
+impl From<v2::FullIncomingNote> for FullIncomingNote {
+    #[inline]
+    fn from(note: v2::FullIncomingNote) -> Self {
+        Self {
+            address_partition: note.address_partition,
+            incoming_note: IncomingNote::from(note.incoming_note),
+        }
+    }
+}
+
+impl TryFrom<FullIncomingNote> for v2::FullIncomingNote {
+    type Error = Error;
+
+    #[inline]
+    fn try_from(note: FullIncomingNote) -> Result<Self, Self::Error> {
+        Ok(Self {
+            address_partition: note.address_partition,
+            incoming_note: note.incoming_note.try_into()?,
+        })
+    }
+}
+
+/// UTXO
+#[cfg_attr(
+    feature = "rpc",
+    derive(Deserialize, Serialize),
+    serde(crate = "manta_util::serde", deny_unknown_fields)
+)]
+#[derive(
+    Clone, Copy, Debug, Decode, Default, Encode, Eq, Hash, MaxEncodedLen, PartialEq, TypeInfo,
+)]
+pub struct Utxo {
+    /// Transparency Flag
+    pub is_transparent: UtxoTransparency,
```

Comment:
nit: maybe `transparency` is better field name, `is_transparent` makes me think it is a `bool`